src/Query/JoinClause.php

```diff
@@ -85,6 +85,12 @@ public function __construct(BaseBuilder $query)
     public function table($table): self
     {
         if (is_string($table)) {
+            if (strpos(strtolower($table), ' as ') !== false) {
+                list($table, $alias) = array_map('trim', preg_split('/\s+as\s+/i', $table));
+
+                $this->as($alias);
+            }
+
```

Comment:
nitpick. I'd rather move it to submethod. Something like
```suggestion
            list($table, $alias) = $this->decomposeToTableAndAlias($table);
            
            if (! is_null($alias)) {
              $this->as($alias);
            }
```

Reply:
Changed to using created submethod.